contracts/provider/assembly/index.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  u128,
+  storage,
+  context,
+  ContractPromiseBatch,
+  PersistentUnorderedMap,
+  logging,
+} from "near-sdk-as";
+import { onlyOwner } from "../../ownable/assembly";
+export {
+  transferOwnership,
+  renounceOwnership,
+  owner,
+} from "../../ownable/assembly";
+
+export const DefaultDivisor = u128.from("416000000000"); // yocto-Near per nano-second
+
+export const deposits = new PersistentUnorderedMap<string, Deposit>("d");
+
+@nearBindgen
+export class Deposit {
```

Comment:
Only data structure we need now. This combined with the above map simplification should help a lot with your state monitoring stuff.